src/lifecycle/state.rs

```diff
@@ -13,21 +13,21 @@ pub trait State: 'static {
     /// the game loop will do its best to still call the update at about 60 TPS.
     ///
     /// By default it does nothing
-    fn update(&mut self, &mut Window) -> Result<()> {
+    fn update(&mut self, window: &mut Window, delta_time: f64) -> Result<()> {
```

Comment:
Indeed it could be. But having to assign a delta time to the window each time instead of just giving it through a parameter seemed more efficient and clean. Also, because there are multiple delta_times (I could even imagine one for events but the user could easily implement that themself, as it has no value to quicksilver), we would have multiple operations more where only a simple pass is required.

Reply:
True, there would be some definite weirdness unless it's a parameter.